### PR TITLE
[PM-12607] Move key rotation & validators to km ownership

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-12607

## 📔 Objective

Move ownership of key-rotation and the validators for key rotation to km ownership.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: test/Api.Test/KeyManagement/Validators/SendRotationValidatorTests.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿using System.Text.Json;
+using Bit.Api.KeyManagement.Validators;
 using Bit.Api.Tools.Models;
 using Bit.Api.Tools.Models.Request;
-using Bit.Api.Tools.Validators;
```
Comment (djsmith85) [inline test/Api.Test/KeyManagement/Validators/SendRotationValidatorTests.cs:4]: ⚠️  Namespace within this file should also be updated

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: test/Api.Test/KeyManagement/Validators/SendRotationValidatorTests.cs (hunk 1 above)
Comment (quexten) [reply test/Api.Test/KeyManagement/Validators/SendRotationValidatorTests.cs:4]: Fixed this, and another instance of forgetting to change the namespace. Thank you!

File: src/Core/KeyManagement/UserKey/IRotateUserKeyCommand.cs [hunk 2]
```diff
@@ -1,9 +1,9 @@
-﻿using Bit.Core.Auth.Models.Data;
-using Bit.Core.Entities;
+﻿using Bit.Core.Entities;
+using Bit.Core.KeyManagement.Models.Data;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.Data.SqlClient;
 
-namespace Bit.Core.Auth.UserFeatures.UserKey;
+namespace Bit.Core.KeyManagement.UserFeatures.UserKey;
```
Comment (djsmith85) [inline src/Core/KeyManagement/UserKey/IRotateUserKeyCommand.cs:6]: ⛏️ Namespace doesn't correspond to file location, due to `UserFeatures` not existing under `KeyManagement`

**Review by djsmith85 — DISMISSED** [review]
Approving for Tools

**Review by gbubemismith — APPROVED** [review]

**Review by BTreston — DISMISSED** [review]

File: test/Api.Test/KeyManagement/Validators/CipherRotationValidatorTests.cs
Comment (mzieniukbw) [inline test/Api.Test/KeyManagement/Validators/CipherRotationValidatorTests.cs:1]: namespace not updated, should be `namespace Bit.Api.Test.KeyManagement.Validators;`

File: test/Api.Test/KeyManagement/Validators/WebauthnLoginKeyRotationValidatorTests.cs
Comment (mzieniukbw) [inline test/Api.Test/KeyManagement/Validators/WebauthnLoginKeyRotationValidatorTests.cs:1]: namespace not updated, should be `namespace Bit.Api.Test.KeyManagement.Validators;`

File: src/Core/KeyManagement/UserKey/Implementations/RotateUserKeyCommand.cs [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 using Bit.Core.Entities;
+using Bit.Core.KeyManagement.Models.Data;
 using Bit.Core.Repositories;
 using Bit.Core.Services;
 using Bit.Core.Tools.Repositories;
 using Bit.Core.Vault.Repositories;
 using Microsoft.AspNetCore.Identity;
 
-namespace Bit.Core.Auth.UserFeatures.UserKey.Implementations;
+namespace Bit.Core.KeyManagement.UserFeatures.UserKey.Implementations;
```
Comment (mzieniukbw) [inline src/Core/KeyManagement/UserKey/Implementations/RotateUserKeyCommand.cs:10]: wrong namespace to the file location, should be:
```suggestion
namespace Bit.Core.KeyManagement.UserKey.Implementations;
```
Or move the file to "UserFeatures" directory.

File: src/Core/KeyManagement/UserKey/IRotateUserKeyCommand.cs (hunk 2 above)
Comment (mzieniukbw) [inline src/Core/KeyManagement/UserKey/IRotateUserKeyCommand.cs:6]: wrong namespace as already pointed out by @djsmith85

**Review by mzieniukbw — CHANGES_REQUESTED** [review]

**Review by mzieniukbw — APPROVED** [review]
LGTM

**Review by ike-kottlowski — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by BTreston — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]

**Pull request merged**